datalad/cmdline/main.py

```diff
@@ -228,7 +228,25 @@ def main(args=None):
         pass
 
     # parse cmd args
-    cmdlineargs = parser.parse_args(args)
+    check_args = args if args else sys.argv
+    # make a cheap, but educated guess whether we might face arguments that
+    # are valid, but unknown to the configured parser
+    # ATM only the 'export' command has an extensible plugin mechanism that
+    # could get us into this situation
+    if 'export' in check_args:
+        # there is a non-zero chance that we will call the 'export' command
+        # call the parser in the most permissive way
+        cmdlineargs, unparsed_args = parser.parse_known_args(args)
```

Comment:
HA -- isn't this a solution to https://github.com/datalad/datalad/issues/446 ? i.e. here, do unconditionally on 'export', but if any unparsed_args, then provide special handling:
```python
 if cmdlineargs.func.__self__.__name__ == 'Export':
   do your thing for export
else:
   spit out --help for the func
```
(that was under condition that it got the func already)